js/views/device_pairing_dialog_view.js

```diff
@@ -16,194 +8,27 @@
-      const waitingForRequestView = this.$('.waitingForRequestView');
-      const requestReceivedView = this.$('.requestReceivedView');
-      const requestAcceptedView = this.$('.requestAcceptedView');
-      const confirmUnpairView = this.$('.confirmUnpairView');
-      if (this.pubKeyToUnpair) {
-        defaultView.hide();
-        requestReceivedView.hide();
-        waitingForRequestView.hide();
-        requestAcceptedView.hide();
-        confirmUnpairView.show();
-        const name = this.getPubkeyName(this.pubKeyToUnpair);
-        this.$('.confirmUnpairView #pubkey').html(name);
-      } else if (!this.isListening) {
-        requestReceivedView.hide();
-        waitingForRequestView.hide();
-        requestAcceptedView.hide();
-        confirmUnpairView.hide();
 
-        const ourPubKey = textsecure.storage.user.getNumber();
-        defaultView.show();
-        const pubKeys = await libloki.storage.getSecondaryDevicesFor(ourPubKey);
-        this.$('#pairedPubKeys').empty();
-        if (pubKeys && pubKeys.length > 0) {
-          this.$('#startPairing').attr('disabled', true);
-          pubKeys.forEach(x => {
-            const name = this.getPubkeyName(x);
-            const li = $('<li>').html(name);
-            if (window.lokiFeatureFlags.multiDeviceUnpairing) {
-              const link = $('<a>')
-                .text('Unpair')
-                .attr('href', '#');
-              link.on('click', () => this.requestUnpairDevice(x));
-              li.append(' - ');
-              li.append(link);
-            }
-            this.$('#pairedPubKeys').append(li);
-          });
-        } else {
-          this.$('#startPairing').removeAttr('disabled');
-          this.$('#pairedPubKeys').append('<li>No paired devices</li>');
-        }
-      } else if (this.accepted) {
-        defaultView.hide();
-        requestReceivedView.hide();
-        waitingForRequestView.hide();
-        requestAcceptedView.show();
-      } else if (this.pubKey) {
-        const secretWords = window.mnemonic.pubkey_to_secret_words(this.pubKey);
-        this.$('.secretWords').text(secretWords);
-        requestReceivedView.show();
-        waitingForRequestView.hide();
-        requestAcceptedView.hide();
-        defaultView.hide();
-      } else {
-        waitingForRequestView.show();
-        requestReceivedView.hide();
-        requestAcceptedView.hide();
-        defaultView.hide();
-      }
+    render() {
+      this.dialogView = new Whisper.ReactWrapperView({
+        className: 'device-pairing-dialog',
+        Component: window.Signal.Components.DevicePairingDialog,
+        props: {
+          i18n,
+          onClose: this.close,
+        },
+      });
+
+      this.$el.append(this.dialogView.el);
+      return this;
     },
+
     close() {
       this.remove();
-      this.qr.clear();
-      if (this.pubKey && !this.accepted) {
-        this.trigger('devicePairingRequestRejected', this.pubKey);
-      }
-      this.trigger('close');
```

Comment:
Shouldn't we keep the trigger of devicePairingRequestRejected? I don't what it does though.

Reply:
Will manage this logic in Whisper wrapper